easybuild/tools/github.py

```diff
@@ -716,6 +738,77 @@ def _easyconfigs_pr_common(paths, start_branch=None, pr_branch=None, target_acco
     return file_info, deleted_paths, git_repo, pr_branch, diff_stat
 
 
+def det_patch_specs(patch_paths, file_info):
+    """ Determine software names for patch files """
+    print_msg("determining software names for patch files...")
+    patch_specs = []
+    for patch_path in patch_paths:
+        soft_name = None
+        patch_file = os.path.basename(patch_path)
+
+        # consider patch lists of easyconfigs being provided
+        for ec in file_info['ecs']:
+            if patch_file in ec['patches']:
+                soft_name = ec.name
+                break
+
+        if soft_name:
+            patch_specs.append((patch_path, soft_name))
+        else:
+            # fall back on scanning all eb files for patches
+            print "Matching easyconfig for %s not found on the first try:" % patch_path,
+            print "scanning all easyconfigs to determine where patch file belongs (this may take a while)..."
+            soft_name = scan_all_easyconfigs(patch_file)
+            if soft_name:
+                patch_specs.append((patch_path, soft_name))
+            else:
+                # still nothing found
+                raise EasyBuildError("Failed to determine software name to which patch file %s relates", patch_path)
+
+    return patch_specs
+
+
+def scan_all_easyconfigs(patch_name):
+    """
+    Scan all easyconfigs in the robot path to determine which software a patch file belongs to
+
+    @param patch_name: name of the patch file
+    """
+    robot_paths = build_option('robot_path')
+    soft_name = ""
+    found = False
+
+    all_ecs = []
+    for robot_path in robot_paths:
+        for (dirpath, _, filenames) in os.walk(robot_path):
+            for fn in filenames:
+                if fn != 'TEMPLATE.eb':
+                    path = os.path.join(dirpath, fn)
+                    rawtxt = read_file(path)
+                    if 'patches' in rawtxt:
+                        all_ecs.append(path)
+
+    nr_of_ecs = len(all_ecs)
+    for idx, filename in enumerate(all_ecs):
+        if found:
+            break
+        rawtxt = read_file(filename)
+        try:
+            ecs = process_easyconfig(filename, validate=False)
+            for ec in ecs:
+                if patch_name in ec['ec']['patches']:
```

Comment:
I just realised this is too simplistic...

The entries in `patches` can also be tuples like `('foo.patch', 2)`, where the `2` is the patch level

So, I'd create an inner function in `find_software_name_for_patch` and use that:

```python
def is_patch_for(patch_name, ec):
    """Check whether specified patch matches any patch in the provided EasyConfig instance."""
    res = False
    for patch in ec['patches']:
        if isinstance(patch, (tuple, list)):
            patch = patch[0]
        if patch == patch_name:
            res = True
            break

    return res
````

and then check with `if is_patch_for(patch_name, ec['ec'])`